docs/proposal/unit_test_framework_zh.md

```diff
@@ -0,0 +1,617 @@
+## **1 背景**
+
+当前，Kmesh目前的测试只有端到端的测试，缺乏对ebpf程序的单元测试框架，我们想要提供不加载整个Kmesh就能对某个ebpf程序进行单元测试的能力，由此能更方便的进行功能测试和覆盖率测试。
+
+## **2 功能设计**
+
+- check()
+- test_init()
+- test_finish()
+- 断言assert()
+- skip机制，根据配置对一些功能的test进行跳过
+
+**设计原则**：对于功能不怎么变化的，在.h里inline实现的，可以采取mock的形式对这些inline函数进行单元测试；对于功能会经常发生变更的，采取端到端的测试。
+
+该框架使用该`BPF_PROG_RUN`功能在内核中运行 eBPF 程序，而无需将它们附加到实际钩子上。
+
+BPF_PROG_RUN用法：
+
+- 一个 eBPF 程序可以定义全局变量，在用户空间测试运行程序之前和之后都可以对其进行修改。
+- 可以使用`BPF_PROG_RUN`命令在用户空间中运行 eBPF 程序，该命令由 libbpf 中的 `bpf_prog_test_run_opts()`函数包装。
+- 一旦将 eBPF 程序编译为对象文件，可以使用 bpftool 生成一个skeleton加载器，该加载器将 eBPF 程序加载到内核，并为用户空间程序提供访问上述全局变量的权限。
+- 最后，可以编写用户态单元测试，在测试之前设置加载的 eBPF 程序的全局变量，并在测试之后读取它们，从而确定 eBPF 程序是否执行了预期的操作。
+
+具体可查看使用示例，链接：https://www.cnxct.com/unit-testing-ebpf/
+
+![Untitled](pics/unit_test_flow.png)
+
+### 2.0 check()
+
+```c
+#define CHECK(progtype, name) __section(progtype "/test/" name "/check")
+```
+
+通过宏定义 `CHECK`，我们可以方便地生成特定格式的 section 名称,可以用这种方式为测试用例定义特定的 section 名称
+
+### **2.1 test_init()**
+
+`test_init`宏用于初始化eBPF测试框架中的一些全局变量和状态。这些变量和状态用于管理测试结果、日志记录和错误处理。它通过设置这些变量，为后续的测试用例提供支持。
+
+关键变量语义：
+
+- `suite_result`：整个测试的结果，初始值为TEST_PASS
+- `test_result_status`：指向当前测试结果状态的指针，初始值设为 0，并使用 `__maybe_unused` 标记以防止编译器警告未使用变量。
+- `suite_result_map` : 存储测试结果以及测试程序相关信息的ebpf map
+- `suite_result_cursor`：指向测试结果存储位置的指针，在`suite_result_map` 查找出对应的项返回赋值给此指针。
+- `test_result_cursor`：指向当前测试结果存储位置的指针，初始值设为 0。
+- `test_result_size`：用于存储测试结果大小的变量。
+
+```c
+#define test_init()               \
+  char suite_result = TEST_PASS;            \
+  __maybe_unused char *test_result_status = 0;        \
+  char *suite_result_cursor;            \
+  {                 \
+    __u32 __key = 0;              \
+    suite_result_cursor =           \
+      test_bpf_map_lookup_elem(&suite_result_map, &__key);\
+    if (!suite_result_cursor) {         \
+      return TEST_ERROR;          \
+    }               \
+  }                 \
+  __maybe_unused char *test_result_cursor = 0;        \
+  __maybe_unused __u16 test_result_size;          \
+  do {
+```
+
+**详细解析**
+
+**1. 定义变量**
+
+```c
+char suite_result = TEST_PASS;
+```
+
+- `suite_result`：表示整个测试套件的结果，初始值设为`TEST_PASS`（测试通过）。
+
+```c
+__maybe_unused char *test_result_status = 0;
+```
+
+- `test_result_status`：指向当前测试结果状态的指针，初始值设为`0`，并使用`__maybe_unused`标记以防止编译器警告未使用变量。
+
+**2. 定义`cursor`变量**
+
+```c
+char *suite_result_cursor;
+```
+
+- `suite_result_cursor`：指向测试结果存储位置的指针。
+
+**3. 查找并初始化`suite_result_cursor`**
+
+```
+{
+  __u32 __key = 0;
+  suite_result_cursor = test_bpf_map_lookup_elem(&suite_result_map, &__key);
+  if (!suite_result_cursor) {
+    return TEST_ERROR;
+  }
+}
+```
+
+- 使用`__key`作为键，从`suite_result_map`中查找值，并将指针赋值给`suite_result_cursor`。
+- 如果找不到值，则返回`TEST_ERROR`，表示测试初始化失败。
```

Comment:
direction return error? Is there no more detailed documentation of the results?
For example, `there is no key`

Reply:
> direction return error? Is there no more detailed documentation of the results? For example, `there is no key`

This parameter corresponds to test_init. { is followed by test_init.